lib/Parser/Parser.yy

```diff
@@ -630,25 +866,68 @@ unique_shared_opt:
     | sharedKw
     ;
 
-literal:
-      boolean_literal
-    | intLit
+variable_literal:
+      intLit
     | floatLit
     | stringLit
+
+literal:
+      boolean_literal
+    | variable_literal
+      {
+        $$ = CREATE_NODE<LiteralExpr>(
+          $1.getLexeme(),
+          CREATE_TYPE<TypeVariableTy>(),
+          LOC($1)
+        );
+        std::cerr << "Parsed type variable literal: " << $1.getLexeme().str() << std::endl;
+      }
     ;
 
 boolean_literal:
-      trueKw
+    | trueKw
```

Comment:
shouldn't be an unintended empty rule, which is why you should use `%empty` when it is intended